metricshub-agent/src/main/java/org/metricshub/web/security/jwt/JwtAuthFilter.java

```diff
@@ -0,0 +1,137 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱
+ */
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import org.metricshub.web.security.SecurityHelper;
+import org.metricshub.web.security.User;
+import org.metricshub.web.service.UserService;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+/**
+ * Filter that processes incoming HTTP requests to authenticate users based on JWT tokens.
+ */
+public class JwtAuthFilter extends OncePerRequestFilter {
+
+	private static final Pattern AUTH_PATH_PATTERN = Pattern.compile("^/auth$");
+
+	private JwtComponent jwtComponent;
+	private UserService userService;
+
+	/**
+	 * Constructor to initialize the filter with necessary components.
+	 *
+	 * @param jwtComponent JWT handling component
+	 * @param userService  User service to retrieve user details
+	 */
+	public JwtAuthFilter(JwtComponent jwtComponent, UserService userService) {
+		this.jwtComponent = jwtComponent;
+		this.userService = userService;
+	}
+
+	@Override
+	protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+		throws ServletException, IOException {
+		// If already authenticated (e.g., by API key), don't override
+		if (SecurityContextHolder.getContext().getAuthentication() != null) {
+			filterChain.doFilter(request, response);
+			return;
+		}
+
+		try {
+			doApiFilter(request);
+		} catch (final Exception ex) {
+			SecurityHelper.writeUnauthorizedResponse(response);
+			return;
+		}
+
+		filterChain.doFilter(request, response);
+	}
+
+	/**
+	 * Get the authentication token, retrieve all the claims, check that user exists
+	 * and finally set the {@link JwtAuthToken} info in the security context holder
+	 *
+	 * @param request HTTP request
+	 */
+	private void doApiFilter(final HttpServletRequest request) {
+		// Skip filtering for POST /auth (login)
+		if (AUTH_PATH_PATTERN.matcher(request.getRequestURI()).matches() && "post".equalsIgnoreCase(request.getMethod())) {

```

Comment:
With:

```java
private static final String AUTH_PATH = "/auth";
```

You can do:

```suggestion
		if (AUTH_PATH.equals(request.getRequestURI()) && "post".equalsIgnoreCase(request.getMethod())) {
```